spackbot/workers.py

```diff
@@ -0,0 +1,185 @@
+import spackbot.comments as comments
+import spackbot.helpers as helpers
+from .auth import authenticate_installation, REQUESTER
+
+logger = helpers.get_logger(__name__)
+
+REDIS_HOST = os.environ.get("REDIS_HOST", "localhost")
+REDIS_PORT = int(os.environ.get("REDIS_PORT", "6379"))
+
+
+class WorkQueue:
+    def __init__(self):
+        logger.info(f"WorkQueue creating redis connection ({REDIS_HOST}, {REDIS_PORT})")
+        self.redis_conn = Redis(host=REDIS_HOST, port=REDIS_PORT)
+        # Name of queue workers use is defined in "workers/entrypoint.sh"
+        self.task_q = Queue(name="tasks", connection=self.redis_conn)
+
+    def get_queue(self):
+        return self.task_q
+
+
+work_queue = WorkQueue()
+
+
+def is_up_to_date(output):
+    """
+    A commit can fail if there are no changes!
+    """
+    return "nothing to commit" in output
+
+
+async def fix_style_task(event, auth_method):
+    """
+    We first retrieve metadata about the pull request. If the request comes
+    from anyone with write access to the repository, we commit, and we commit
+    under the identity of the original person that opened the PR.
+    """
+    logger.debug("fix_style_task, auth_method = ")
+    logger.debug(auth_method)
+
+    token = None
+
+    if "oauth_token" in auth_method:
+        token = auth_method["oauth_token"]
+    elif "installation_id" in auth_method:
+        token = await authenticate_installation(auth_method["installation_id"])
+
+    if not token:
+        logger.error("fix_style_task() unable to authenticate installation")
+        return
+
+    async with aiohttp.ClientSession() as session:
+        gh = gh_aiohttp.GitHubAPI(session, REQUESTER, oauth_token=token)
+
+        pr_url = event.data["issue"]["pull_request"]["url"]
+
+        pr = await gh.getitem(pr_url)
+
+        logger.debug("GitHub PR")
+        logger.debug(pr)
+
+        # Get the sender of the PR - do they have write?
+        sender = event.data["sender"]["login"]
+        repository = event.data["repository"]
+        collaborators_url = repository["collaborators_url"]
+
+        logger.debug(
+            f"sender = {sender}, repo = {repository}, collabs_url = {collaborators_url}"
+        )
+
+        # If they don't have write, we don't allow the command
+        if not await helpers.found(
+            gh.getitem(collaborators_url, {"collaborator": sender})
+        ):
+            msg = f"Sorry {sender}, I cannot do that for you. Only users with write can make this request!"
+            await gh.post(event.data["issue"]["comments_url"], {}, data={"body": msg})
+            return
+
+        # Tell the user the style fix is going to take a minute or two
+        message = "Let me see if I can fix that for you!"
+        await gh.post(event.data["issue"]["comments_url"], {}, data={"body": message})
+
+        # Get the username of the original committer
+        user = pr["user"]["login"]
+
+        # We need the user id if the user is before July 18. 2017
```

Comment:
Link to https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address, which has the note about why